pilot/cmd/pilot-agent/status/app/probe.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	// ProbeMapEnvName is the name of the command line flag for pilot agent to pass app prober config.
+	// The json encoded string to pass app HTTP probe information from injector(istioctl or webhook).
+	// For example, --ISTIO_KUBE_APP_PROBERS='{"/app-health/httpbin/livez":{"path": "/hello", "port": 8080}.
+	// indicates that httpbin container liveness prober port is 8080 and probing path is /hello.
+	// This environment variable should never be set manually.
+	ProbeMapEnvName = "ISTIO_KUBE_APP_PROBERS"
+)
+
+var (
+	appProberPatternStr = `^/app-health/[^/]+/(livez|readyz)$`
+	appProberPattern    = regexp.MustCompile(appProberPatternStr)
+)
+
+// ProbeURLPath the URL path for a pilot agent prober.
+type ProbeURLPath string
```

Comment:
This is only used for app prober not the pilot agent itself, right?

Reply:
Correct